compatibility-verifier/compCheck.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# A script that does rolling upgrade of Pinot components
+# from one version to the other given 2 commit hashes. It first builds  
+# Pinot in the 2 given directories and then upgrades in the following order:
+# Controller -> Broker -> Server
+
+# get usage of the script 
+function usage() {
+  command=$1
+  if [[ "$command" =~ compCheck.sh$ ]] ; then
+    echo "Usage: $command olderCommit newerCommit [workingDir]"
+  fi
+  exit 1
+}
+
+# cleanup the temporary directory when exiting the script
+function cleanup() {
+  if [ -n "$tmpDir" ] && [ -d "$tmpDir" ]; then
```

Comment:
For now, let us just echo that this needs to be cleaned up.
In some cases, we would want to keep the dir to see the logs/results, so it may be useful to keep it.
Also, I would prefer that we set a variable called TMP_DIR up front to the results of mktemp. The use of `TMP_DIR` indicates that we don't modify it -- just a convention.

Reply:
So do you mean just renaming tmpDir to TMP_DIR?

Reply:
just move the assignment to `tmpDir` to be an assignment to `TMP_DIR` and move it to the top of the file. It does not create the dir anyway, only assigns a dir name. We can create the dir if needed (if working dir is not supplied)